src/helics/application_api/helicsTypes.cpp

```diff
@@ -178,20 +163,67 @@ static const std::unordered_map<std::string, data_type> typeMap{
     {"", data_type::helics_any},
     {"all", data_type::helics_any}};
 
-data_type getTypeFromString(const std::string& typeName)
+static const std::unordered_map<std::string, data_type> demangle_names{
```

Comment:
The manged/demangled names are only used within a federate and never get sent to another federate, right?

Reply:
The only place this is used, is when a user enters a type, via a string, This should detect if it is one of the primary types.   I should double check that it actually maps it to a common string though.  If not that might be something we should do in HELICS 3.

Reply:
It is definitely possible to change the string as it comes in to some stable type, but I am a little hesitant to do that right now, since it could mean that if the user queries the type they put in, they would get something else back.    I am not sure we want that.

Reply:
I added a clean function for the types, but it will only change them if they are matched through the demangle map,  if they match in the static map, they will stay the same.

Reply:
It will do this for input and publications